### PR TITLE
Initiative order

### PR DESCRIPTION
**Pull request by Westlanderz**
This PR will rework the combat system to work on a prep phase and executing phase with initiative.

### CODE REVIEW

File: Assets/Scripts/Core/Controllers/GameController.cs [hunk 1]
```diff
@@ -16,9 +16,11 @@ public class GameController : MonoBehaviour {
 
 		[SerializeField] private MovementTilemapVisual movementTilemapVisual_;
 		[SerializeField] private MovementTilemapVisual arrowTilemapVisual_;
+		[SerializeField] private MovementTilemapVisual seletorTilemapVisual_;
```
Comment (drfdaniel) [inline Assets/Scripts/Core/Controllers/GameController.cs:19]: seletor? you mean selector?

File: Assets/Scripts/Core/Controllers/SaveController.cs.meta [hunk 2]
```diff
@@ -1,10 +1,10 @@
 fileFormatVersion: 2
-guid: 8067fc8595231f6d59eb94b290583f29
+guid: 792253aedb5774b90bd18021a4497e14
 MonoImporter:
   externalObjects: {}
   serializedVersion: 2
   defaultReferences: []
-  executionOrder: 1
+  executionOrder: 0
```
Comment (Dennisbonke) [inline Assets/Scripts/Core/Controllers/SaveController.cs.meta:7]: Why was this changed back? Is this tested? Intended?

File: Assets/Scripts/Core/Interactables/Vent.cs [hunk 3]
```diff
@@ -13,9 +13,9 @@ public class Vent : MonoBehaviour, IInteractable {
 			GameController.Instance.GetGrid().GetGridObject(transform.position).SetInteractable(this);
 		}
 
-		public void Interact() {
+		public void Interact(CoreUnit unit) {
 			GridCombatSystem combatInstance = GridCombatSystem.Instance;
-			CoreUnit player = combatInstance.GetActiveUnit();
+			CoreUnit player = unit;
```
Comment (Dennisbonke) [inline Assets/Scripts/Core/Interactables/Vent.cs:18]: Shouldn’t we just replace player with unit? Or call the incoming parameter (for this function that is)?

File: Assets/Scripts/Core/Interactables/Vent.cs [hunk 4]
```diff
@@ -32,13 +32,11 @@ public class Vent : MonoBehaviour, IInteractable {
 				unit = player
 			};
 			combatInstance.OnUnitActionPointsChanged?.Invoke(this, unitEvent);
-			combatInstance.UpdateValidMovePositions();
```
Comment (Dennisbonke) [inline Assets/Scripts/Core/Interactables/Vent.cs:35]: I assume you tested this?

**Review by drfdaniel — DISMISSED** [review]

File: Assets/Scripts/Core/Controllers/GameController.cs (hunk 1 above)
Comment (Westlanderz) [reply Assets/Scripts/Core/Controllers/GameController.cs:19]: Well yes, but thought I fixed it already.

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
Some questions from a partial first look. I fully expect more to pop up during in-depth reviewing.

File: Assets/Scripts/Core/Interactables/Vent.cs (hunk 4 above)
Comment (Westlanderz) [reply Assets/Scripts/Core/Interactables/Vent.cs:35]: Yes, as it ends the turn it does not need to be here because the new turn will update it.

File: Assets/Scripts/Core/Interactables/Vent.cs (hunk 3 above)
Comment (Westlanderz) [reply Assets/Scripts/Core/Interactables/Vent.cs:18]: This was a lazy solution so did not rename it.

File: Assets/Scripts/Core/Controllers/SaveController.cs.meta (hunk 2 above)
Comment (Westlanderz) [reply Assets/Scripts/Core/Controllers/SaveController.cs.meta:7]: Unity changed this so it sould be fine.

File: Assets/Scripts/Core/Interactables/Vent.cs (hunk 3 above)
Comment (Dennisbonke) [reply Assets/Scripts/Core/Interactables/Vent.cs:18]: Fair enough.

File: Assets/Scripts/Core/Interactables/Vent.cs (hunk 4 above)
Comment (Dennisbonke) [reply Assets/Scripts/Core/Interactables/Vent.cs:35]: Ah I understand, this looks good to me then.

File: Assets/Scripts/Core/Controllers/SaveController.cs.meta (hunk 2 above)
Comment (Dennisbonke) [reply Assets/Scripts/Core/Controllers/SaveController.cs.meta:7]: Considering I did this change manually, I need to test this locally before I can approve of this.

Comment (Westlanderz) [reply Assets/Scripts/Core/Controllers/SaveController.cs.meta:7]: But it is Unity best practice to accept all Unity generated files, and `.meta` files are generated by Unity.

Comment (Dennisbonke) [reply Assets/Scripts/Core/Controllers/SaveController.cs.meta:7]: Yes, but you can change the execution order from the editor (which I did) and that change is reverted. I need to test this for breakage before approving this PR.

Comment (Westlanderz) [reply Assets/Scripts/Core/Controllers/SaveController.cs.meta:7]: I tested this within the fixed mapeditor scene and the saving and loading still works fine.

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
Code makes sense to me, but a few questions and nitpicks remain.

File: Assets/Scripts/Core/Controllers/GridCombatSystem.cs [hunk 5]
```diff
@@ -0,0 +1,688 @@
...
+					}
+					if((node.parent.parent.xPos == node.parent.xPos && node.parent.xPos > x 
+						&& node.parent.parent.yPos < node.parent.yPos && node.parent.yPos == y)
+						|| (node.parent.parent.xPos < node.parent.xPos && node.parent.xPos == x 
+						&& node.parent.parent.yPos == node.parent.yPos && node.parent.yPos > y)) {
+						arrowMap.SetRotation(node.parent.xPos, node.parent.yPos, -90f);
+						arrowMap.SetTilemapSprite(node.parent.xPos, node.parent.yPos, MovementTilemap.TilemapObject.TilemapSprite.ArrowCorner);
+					}
+				}
+				if(actions.Count != 0 && node.parent != null) {
```
[the remark this answers is not in the document]
Comment (Westlanderz) [reply Assets/Scripts/Core/Controllers/GridCombatSystem.cs:678]: This is for the arrows later on when overlapping, have not made it yet but is fine to leave here.

File: Assets/Scripts/Core/Utils/PrefabStatus.cs.meta [hunk 6]
```diff
@@ -1,10 +1,10 @@
 fileFormatVersion: 2
-guid: 70959477dc1fa5c24a320471a31243c3
+guid: 47c4551d7e41a810a92c2893305323ee
 MonoImporter:
   externalObjects: {}
   serializedVersion: 2
   defaultReferences: []
-  executionOrder: -2
+  executionOrder: 0
```
[the remark this answers is not in the document]
Comment (Westlanderz) [reply Assets/Scripts/Core/Utils/PrefabStatus.cs.meta:7]: Why was it at -2 in the first place, seems to me that it is executing just fine.

File: Assets/Scripts/Core/Controllers/GridCombatSystem.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (Dennisbonke) [reply Assets/Scripts/Core/Controllers/GridCombatSystem.cs:678]: Alright then, good to know.

File: Assets/Scripts/Player/Movement/MovePositionPathfinding.cs [hunk 7]
```diff
@@ -1,6 +1,7 @@
 ﻿using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Threading.Tasks;
```
[the remark this answers is not in the document]
Comment (Westlanderz) [reply Assets/Scripts/Player/Movement/MovePositionPathfinding.cs:4]: That was added for use of Tasks which are a sort of return type for async functions. Does no harm to leave it in.

File: Assets/Scripts/Core/Utils/PrefabStatus.cs.meta (hunk 6 above)
[the remark this answers is not in the document]
Comment (Dennisbonke) [reply Assets/Scripts/Core/Utils/PrefabStatus.cs.meta:7]: If it still works it is fine to me.

File: Assets/Scripts/Player/Movement/MovePositionPathfinding.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (Dennisbonke) [reply Assets/Scripts/Player/Movement/MovePositionPathfinding.cs:4]: Gotcha.

File: Assets/Scripts/Core/Utils/PrefabStatus.cs.meta (hunk 6 above)
[the remark this answers is not in the document]
Comment (Westlanderz) [reply Assets/Scripts/Core/Utils/PrefabStatus.cs.meta:7]: Please test it yourself as well, just to make sure.

**Review by Dennisbonke — APPROVED** [review]
LGTM now
